### PR TITLE
Join timeline with sections

### LINKED ISSUE #722: Join timeline with sections
Wrap section into comment card and display them in feed form

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Feature

## Related issues

Resolve #722

### CODE REVIEW

File: src/components/InterviewSectionListItem.tsx [hunk 1]
```diff
@@ -1,56 +1,60 @@
...
+import s from './CommentView/CommentView.module.css';
+import { Circle } from './Circle';
+import { Link } from './Link';
+import { CardHeaderSection } from './CardHeaderSection';
+import { MarkdownRenderer } from './MarkdownRenderer/MarkdownRenderer';
 
 interface Props {
     section: SectionWithSectionTypeAndInterviewerAndSolutionsRelations;
     interview: InterviewWithRelations;
+    className?: string;
```
Comment (asabotovich) [inline src/components/InterviewSectionListItem.tsx:22]: we can replace this with `extends HTMLAttributes<HTMLDivElement>`

File: src/components/CardHeaderSection.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    interview: Interview & { candidateSelectedSectionId: number | null };
+    menu?: DropdownMenuItem[];
+}
+
+const StyledWrap = styled.div`
+    display: inline-flex;
+    gap: ${gapXs};
+    align-items: center;
+    white-space: nowrap;
+`;
```
Comment (asabotovich) [inline src/components/CardHeaderSection.tsx:30]: why do we use styled in new components?

File: src/components/CardHeaderSection.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    return (
+        <StyledWrap>
+            <Text size="l" weight="bold">
+                {link ? <Link href={link}>{section.sectionType.title}</Link> : section.sectionType.title}
+                {menu && <TitleMenu items={menu} />}
+            </Text>
+
+            <InlineDot />
+            {sectionName.length > 0 && (
+                <>
```
Comment (asabotovich) [inline src/components/CardHeaderSection.tsx:48]: remove fragment

File: src/components/CardHeaderSection.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { CandidateSelectedSectionBadge } from './CandidateSelectedSectionBadge';
+import { Link } from './Link';
+import { TitleMenu } from './TitleMenu';
+import { DropdownMenuItem } from './TagFilterDropdown';
+
+interface CardHeaderSectionProps extends React.HTMLAttributes<HTMLDivElement> {
+    name: string;
+    timeAgo: string;
+    section: SectionWithInterviewerRelation;
+    interview: Interview & { candidateSelectedSectionId: number | null };
```
Comment (asabotovich) [inline src/components/CardHeaderSection.tsx:21]: InterviewWithRelations ? (like in InterviewSectionListItem?)

File: src/components/Interview/Interview.tsx [hunk 5]
```diff
@@ -171,21 +171,22 @@ export const Interview: VFC<InterviewProps> = ({ interview, sectionTypes, reject
...
-                        <Text size="xxl" style={{ marginTop: 30 }}>
-                            {tr('Interview sections')}
-                        </Text>
-                        {interview.sections.map((section) => (
-                            <InterviewSectionListItem key={section.id} section={section} interview={interview} />
-                        ))}
-                    </>
-                )}
-                <StyledTitle size="xl">{tr('Comments')}</StyledTitle>
+                <StyledTitle size="xl">Activity Feed</StyledTitle>
```
Comment (tsumo) [inline src/components/Interview/Interview.tsx:174]: translation

File: src/modules/interviewTypes.ts [hunk 6]
```diff
@@ -105,14 +105,19 @@ export type SectionWithSectionTypeAndInterviewerAndSolutionsRelations = Section
...
 
+export type ActivityFeedItem =
+    | { type: 'section'; value: SectionWithSectionTypeAndInterviewerAndSolutionsRelations }
+    | { type: 'comment'; value: CommentWithUserAndReaction };
+
 export type InterviewWithRelations = InterviewWithHireStreamRelation & {
     creator: User;
     candidate: Candidate;
-    sections: SectionWithSectionTypeAndInterviewerAndSolutionsRelations[];
+    sections: SectionWithSectionTypeAndInterviewerAndSolutionsRelations[]; // TODO: remove??
```
Comment (tsumo) [inline src/modules/interviewTypes.ts:115]: Why remove them? You still need them to construct activity feed

File: src/modules/interviewTypes.ts [hunk 7]
```diff
@@ -105,14 +105,19 @@ export type SectionWithSectionTypeAndInterviewerAndSolutionsRelations = Section
...
     candidateSelectedSection: Section | null;
     cv: Attach | null;
     restrictedUsers?: User[];
-    comments: CommentWithUserAndReaction[] | undefined;
+    comments: CommentWithUserAndReaction[] | undefined; // TODO: remove ??
```
Comment (tsumo) [inline src/modules/interviewTypes.ts:119]: same

**Review by tsumo — CHANGES_REQUESTED** [review]

**Comment by Katrin-kudryash** [comment]
Untitled 3

**Comment by Katrin-kudryash** [comment]
Untitled 3
Untitled 4

**Comment by Katrin-kudryash** [comment]
Untitled 4
Untitled 3

**Review by tsumo — APPROVED** [review]

File: src/components/Interview/Interview.tsx [hunk 8]
```diff
@@ -171,21 +171,22 @@ export const Interview: VFC<InterviewProps> = ({ interview, sectionTypes, reject
...
-                )}
-                <StyledTitle size="xl">{tr('Comments')}</StyledTitle>
+                <StyledTitle size="xl">{tr('Activity Feed')}</StyledTitle>
+
                 <div className={s.InterviewCommentWrapper}>
-                    {interview.comments?.map((comment) => (
-                        <Comment key={`comment - ${comment.id}`} comment={comment} />
-                    ))}
+                    {interview.activityFeed.length > 0 &&
+                        interview.activityFeed.map((item) =>
```
Comment (asabotovich) [inline src/components/Interview/Interview.tsx:178]: use nullable

File: src/components/InterviewSectionListItem.tsx [hunk 9]
```diff
@@ -1,56 +1,69 @@
...
-            chips={
-                <>
-                    <Badge size="xl" color={SectionStatusTagPalette[sectionChip]}>
-                        {sectionChip}
-                    </Badge>
+    const SectionStatusColors: Record<SectionStatus, string> = {
+        [SectionStatus.HIRE]: s.CardInfoSectionHire,
+        [SectionStatus.NO_HIRE]: s.CardInfoSectionNoHire,
+        [SectionStatus.NEW]: s.CardInfoSectionNew,
+    };
```
Comment (asabotovich) [inline src/components/InterviewSectionListItem.tsx:39]: You can keep out of component

**Review by asabotovich — APPROVED** [review]

**Pull request merged**